pages/web-literacy/WebLiteracyPage.jsx

```diff
@@ -1,44 +1,314 @@
+    var state = {};
+    if (this.state.topic !== topic || this.state.webLitSkill !== webLitSkill) {
+      state.topic = topic;
+      state.webLitSkill = webLitSkill;
+      this.setState(state);
+    }
+  },
+  componentDidUpdate: function() {
+    this.updateMapNavState();
+  },
+  componentDidMount: function() {
+    this.updateMapNavState();
+  },
+  getInitialState: function() {
+    var filter = {};
+    Object.keys(categories).map(function(cat) {
+      filter[categories[cat]] = false;
+    });
+    return {
+      topic: "",
+      webLitSkill: "",
+      filter: filter
+    };
+  },
+  hasWebLitSkillIn: function(webLitSkills) {
+    var webLitSkill = this.state.webLitSkill;
+    if (webLitSkill && webLitSkills.indexOf(webLitSkill) !== -1) {
+      return true;
+    }
+  },
+  hasMatchingWebLitSkillIn: function(webLitSkills) {
+    var selectedTopic = this.state.topic;
+    if (!selectedTopic || this.state.webLitSkill) {
+      return false;
+    }
+    var topicWebLitSkills = Object.keys(weblitdata[selectedTopic]);
+    return webLitSkills.some(function(webLitSkill) {
+      return topicWebLitSkills.indexOf(webLitSkill) !== -1;
+    });
+  },
+  hasMatching21CSkillIn: function(skills21C) {
+    var state = this.state;
+    var hasCategory = this.hasCategory;
+    return skills21C.some(function(skill21C) {
+      return !state.filter[categories[skill21C]] && hasCategory(skill21C, state.topic, state.webLitSkill);
+    });
+  },
+  hasCategory: function(cat, selectedVerb, selectedWebLitSkill) {
+    var cat = categories[cat];
+
+    if (!selectedVerb) {
+      return true;
+    }
 
-        <section className="intro">
+    if (!selectedWebLitSkill) {
+      return Object.keys(weblitdata[selectedVerb]).some(function(item) {
+        return weblitdata[selectedVerb][item].indexOf(cat) !== -1;
+      });
+    }
+
+    return weblitdata[selectedVerb][selectedWebLitSkill].indexOf(cat) !== -1;
+  },
+  renderActivities: function() {
+    var activities = [];
+    activitydata.forEach(function(activity, index) {
+      if ((this.hasWebLitSkillIn(activity.webLitSkills) || this.hasMatchingWebLitSkillIn(activity.webLitSkills)) && this.hasMatching21CSkillIn(activity.skills)) {
+        activities.push(
+          <div className="activity-item" key={index}>
+            <Activity
+              selectedTopic={this.state.topic}
+              selectedWebLitSKill={this.state.webLitSkill}
+              topic={activity.topic}
+              name={activity.name}
+              src1x={activity.imgSrc1x}
+              src2x={activity.imgSrc2x}
+              content={activity.content}
+              duration={activity.duration}
+              difficulty={activity.difficulty}
+              webLitSkills={activity.webLitSkills}
+              skills={activity.skills}
+              href={activity.href}
+            />
+          </div>
+        );
+      }
+    }.bind(this));
+
+    if (!activities.length) {
+      return null;
+    } else {
```

Comment:
remember, you don't need an `else` keyword if your `if` case returns. If you're still in the function after the `if` block, you're automatically in the "else" case.